vulcanai2/models/basenetwork.py

```diff
@@ -89,14 +96,46 @@ def __init__(self, name, dimensions, config, save_path=None, input_network=None,
             validation_accuracy=[]
         )
 
-        # self._itr = 0 #TODO: ?
+        self.in_dim = self._dimensions
+        if self._input_networks:
+            for i, in_net in enumerate(self._input_networks):
+                if in_net.__class__.__name__ == "ConvNet":
+                    if in_net.conv_flat_dim != self.in_dim:
+                        self.in_dim[i] = self.get_flattened_size(in_net)
```

Comment:
we shouldn't instantly flatten a ConvNet input, especially if it's incoming to another ConvNet object. Since this is in basenetwork, we should keep it very general and not specific to any inherited-child object (issues with circular object refs)